downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/MolecularFunctionAnnotationBuilder.java

```diff
@@ -150,27 +153,28 @@ private static void processProteins(Set<GKInstance> proteinInstances, GKInstance
      * @param reactionInst -- GKInstance, parent reaction instance that protein/catalyst comes from.
      * @throws Exception -- MySQLAdaptor exception.
      */
-    private static void getGOMolecularFunctionLine(GKInstance catalystInst, GKInstance referenceEntityInst, String taxonIdentifier, GKInstance reactionInst) throws Exception {
+    private static String generateGOMolecularFunctionLine(GKInstance catalystInst, GKInstance referenceEntityInst, String taxonIdentifier, GKInstance reactionInst) throws Exception {
         List<String> pubMedIdentifiers = new ArrayList<>();
         for (GKInstance literatureReferenceInst : (Collection<GKInstance>) catalystInst.getAttributeValuesList(ReactomeJavaConstants.literatureReference)) {
             pubMedIdentifiers.add(PUBMED_IDENTIFIER_PREFIX + literatureReferenceInst.getAttributeValue(ReactomeJavaConstants.pubMedIdentifier).toString());
         }
         GKInstance activityInst = (GKInstance) catalystInst.getAttributeValue(ReactomeJavaConstants.activity);
+        String goaLine = null;
         if (!GOAGeneratorUtilities.isProteinBindingAnnotation(activityInst.getAttributeValue(ReactomeJavaConstants.accession).toString())) {
```

Comment:
There's a fair bit going on in this line.  Is it possible for the `isProteinBindingAnnotation` method to take the `activityInst` as a parameter directly and do the processing in that utility method instead of its accession as String object?

Reply:
Good point, I moved the accession retrieval to the method